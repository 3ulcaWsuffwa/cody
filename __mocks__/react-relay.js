const ReactRelay = require("react-relay");

module.exports = {
  ...ReactRelay,
  // Mock createFragmentContainer HOC so it just returns the inner Component
  createFragmentContainer: Component => Component
};
