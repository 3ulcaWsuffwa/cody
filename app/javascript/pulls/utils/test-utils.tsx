import React from "react";
import { render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import { MemoryRouter } from "react-router";

// eslint-disable-next-line react/prop-types
const TestProvider = ({ children }) => {
  return (
    <IntlProvider locale="en" timeZone="UTC">
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </IntlProvider>
  )
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
function customRender(ui, options?) {
  return render(ui, { wrapper: TestProvider, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
