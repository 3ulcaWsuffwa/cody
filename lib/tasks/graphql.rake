namespace :graphql do
  desc "Export the GraphQL schema"
  task schema: :environment do
    File.write(
      Rails.root.join("schema.graphql"),
      GraphQL::Schema::Printer.print_schema(CodySchema)
    )
  end

  desc "Run Relay compiler"
  task relay: :schema do
    sh "bin/yarn relay"
  end
end
