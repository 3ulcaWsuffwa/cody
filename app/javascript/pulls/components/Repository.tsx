import React from "react";
import { Link } from "react-router-dom";
import { createFragmentContainer, graphql } from "react-relay";
import type { Repository_repository } from "./__generated__/Repository_repository.graphql";
import { ChevronRight, Folder } from "react-feather";

function Repository({
  repository,
}: {
  repository: Repository_repository;
}): JSX.Element {
  return (
    <li>
      <Link
        to={`/repos/${repository.owner}/${repository.name}`}
        className="block hover:bg-gray-50"
      >
        <div className="flex items-center px-4 py-4 sm:px-6">
          <div className="min-w-0 flex-1 flex items-center">
            <div className="flex-shrink-0">
              <Folder />
            </div>
            <div className="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4">
              <div>
                <p className="text-sm font-medium truncate">{`${repository.owner}/${repository.name}`}</p>
              </div>
            </div>
            <div>
              <ChevronRight />
            </div>
          </div>
        </div>
      </Link>
    </li>
    // <div className="level box">
    //   <div className="level-left code">
    //     <div className="level-item">
    //       <strong>{`${repository.owner}/${repository.name}`}</strong>
    //     </div>
    //   </div>
    //   <div className="level-right">
    //     <div className="level-item">
    //       <Link
    //         to={`/repos/${repository.owner}/${repository.name}`}
    //         className="button"
    //         title={`${repository.owner}/${repository.name}`}
    //       >
    //         <MoreHorizontal />
    //       </Link>
    //     </div>
    //   </div>
    // </div>
  );
}

export default createFragmentContainer(Repository, {
  repository: graphql`
    fragment Repository_repository on Repository {
      id
      owner
      name
    }
  `,
});
