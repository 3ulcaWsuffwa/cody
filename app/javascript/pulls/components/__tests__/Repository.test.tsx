import React from "react";
import { render } from "test-utils";
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils";
import { QueryRenderer, graphql, Environment } from "react-relay";
import Repository from "../Repository";
import { RepositoryTestSnapshotQuery } from "./__generated__/RepositoryTestSnapshotQuery.graphql";

const TestComponent = ({ environment }: { environment: Environment }) => (
  <QueryRenderer<RepositoryTestSnapshotQuery>
    environment={environment}
    query={graphql`
      query RepositoryTestSnapshotQuery @relay_test_operation {
        viewer {
          repositories(first: 10) {
            edges {
              node {
                ...Repository_repository
              }
            }
          }
        }
      }
    `}
    variables={{}}
    render={({ error, props }) => {
      if (error) {
        return error.message;
      } else if (props) {
        return (
          <Repository repository={props.viewer.repositories.edges[0].node} />
        );
      } else {
        return "Loading";
      }
    }}
  />
);

let environment;
beforeEach(() => {
  environment = createMockEnvironment();
});

test("Repository snapshot test", () => {
  const renderedComponent = render(<TestComponent environment={environment} />);

  environment.mock.resolveMostRecentOperation((operation) =>
    MockPayloadGenerator.generate(operation)
  );

  const fragment = renderedComponent.asFragment();
  expect(fragment).toMatchSnapshot();
});
