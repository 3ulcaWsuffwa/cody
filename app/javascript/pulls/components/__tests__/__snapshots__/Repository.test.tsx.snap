// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository snapshot test 1`] = `
<DocumentFragment>
  <li>
    <a
      class="block hover:bg-gray-50"
      href="/repos/<mock-value-for-field-\\"owner\\">/<mock-value-for-field-\\"name\\">"
    >
      <div
        class="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          class="mix-w-0 flex-1 flex items-center"
        >
          <div
            class="flexh-shrink-0"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 19a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h5l2 3h9a2 2 0 0 1 2 2z"
              />
            </svg>
          </div>
          <div
            class="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4"
          >
            <div>
              <p
                class="text-sm font-medium truncate"
              >
                &lt;mock-value-for-field-"owner"&gt;/&lt;mock-value-for-field-"name"&gt;
              </p>
            </div>
          </div>
          <div>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
